test/Pool.test.js

```diff
@@ -475,6 +476,26 @@ describe('Pool', function () {
         });
   });
 
+  it('should respect signal and emit event from worker', function (done) {
```

Comment:
Is the abort signal working right now for rejections, `timeout` and `cancel` cases? Can you add a unit test for each of these three cases?

Reply:
At this time the pr does not have support for trigger the `abort` when `timeout` or `cancel` is invoked. I am close to having this part complete but have an edge case I am working.